src/map.cpp

```diff
@@ -4723,8 +4721,8 @@ bool map::open_door( Creature const &u, const tripoint_bub_ms &p, const bool ins
                            "open_door", ter.id.str() );
             ter_set( p, ter.open );
 
-            if( u.has_trait( trait_SCHIZOPHRENIC ) && u.is_avatar() &&
-                one_in( 50 ) && !ter.has_flag( ter_furn_flag::TFLAG_TRANSPARENT ) ) {
+            if( u.is_avatar() && u.as_avatar()->schizo_symptoms( 50 ) &&
+                !ter.has_flag( ter_furn_flag::TFLAG_TRANSPARENT ) ) {
```

Comment:
The previously inconsistent case is here. It didn't check for `effect_took_thorazine` presumably because `Creature` doesn't have `has_effect`. Since we've confirmed `is_avatar` immediately prior, `as_avatar` to get access to `has_effect` or `schizo_symptoms` is safe.